vignettes/quickstart.Rmd

```diff
@@ -0,0 +1,286 @@
+imputed_dfs[[1]]
+imputed_dfs[[2]]
+```
+
+Note that in the case of `method_condmean()` the first imputed dataset will always be the
+"full" dataset containing all the original patients. In the case of `type=jackknife` the
+other datasets will all contain n-1 patients, whilst in the case of `type=bootstrap` each
+dataset will contain a bootstrap sample of patients.
+
+
+## Analyse
+
+The next step is to run our analysis on each of our imputed datasets. This is done by defining
+an analysis function and then getting `analyse()` to map this function across each of our
+imputed datasets. For this vignette will we use the `ancova()` function provided by the rbmi
+package which fits an ancova model independently for each visit returning the treatment 
+effect and least square means for each visit. 
+
+```{r}
+anaObj <- analyse(
+    imputeObj,
+    ancova,
+    vars = set_vars(
+        subjid = "id",
+        outcome = "outcome",
+        visit = "visit",
+        group = "group",
+        covariates = c("age", "sex")
+    )
+)
+anaObj
+```
+
+Note that, similar to draws, the ancova function needs use to use the `set_vars()`
+function to tell it the names of the key variables within the data as well as which
+covariates we want to include in our analysis model.
+
+Additionally we can use the `delta` argument of `analyse()` to perform a tipping point
+analysis. Essentially this works by us specifying a data.frame that contains the amount
+of bias to add to the outcome variable. This must be specified separately for each patient
+at each visit. This data.frame must contain the columns `vars$visit`, `vars$subjid` and `delta`.
+
+It is appreciated that this is potentially tedious to specify and as such two helper
+functions, `delta_template()` has been provided to simplify this process. In particular
+`delta_template()` provides a shell data.frame where the bias is set to 0 for all
+patients but where many additional meta-variables are provided in order to support
+us defining our own logic.  
+
+For example lets say we want to have a rule that we add a bias of 5 to all observations
+missing observations in the control arm. That could then be implemented as follows:
+
+```{r}
+## For reference show the additional meta variables provided
+delta_template(imputeObj) %>% as_tibble()
+
+delta_df <- delta_template(imputeObj) %>%
+    as_tibble() %>% 
+    mutate(delta = if_else(group == "A" & is_missing , 5, 0)) %>% 
+    select(id, visit, delta)
+    
+delta_df
+
+anaObj_delta <- analyse(
+    imputeObj,
+    ancova,
+    delta = delta_df,
+    vars = set_vars(
+        subjid = "id",
+        outcome = "outcome",
+        visit = "visit",
+        group = "group",
+        covariates = c("age", "sex")
+    )
+)
+```
+
+## Pool
+
+Finally, the `pool()` function can be used to summarise all of our analysis results down 
+into a single statistic including standard error, confidence intervals and a p-value for
+the hypothesis test that $H_0 = 0$.
```

Comment:
We could add something like:

The pooling method is automatically derived based on the approach adopted:

- if `method_bayes()` or `method_approxbayes()` then pooling and inference are based on Rubin's rules.
- if `method_condmean(type = "bootstrap")` then inference is based on either normal approximation of the treatment effect distribution (`pool(type = "normal")`) or on percentiles (`pool(type = "percentile")`).
- if `method_condmean(type = "jackknife")` then inference is based on normal approximation.

In this case, since we are using a standard Bayesian multiple imputation procedure, the `pool()` function will automatically use Rubin's rules.

Reply:
done